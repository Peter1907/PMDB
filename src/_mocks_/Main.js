/* eslint-disable*/
import React from 'react';
import * as Scroll from 'react-scroll';
import s from './Main.module.css';
import store from './store';

const ScrollLink = Scroll.Link;

export default function Main() {
  const logo = ('./assets/logo02.png');

  const Data = store.topMovies;
  const images = Data.slice(0, 2);

  return (
    <div className={s.mainContainer}>
      <div className={s.sliderContainer}>
        <div className={s.imgContainer}>
          {images.map((item, id) => (
            <img key={id} id={`image-${id + 1}`} src={item.image} alt={images.title} />
          ))}
        </div>
        <div className={s.dotsContainer}>
          {images.map((item, id) => (
            <a key={id} href={`image-${id}`}> </a>
          ))}
        </div>
      </div>
      <div className={s.gradient} />
      <div className={s.textBox}>
        <div className={s.title}>
          <img src={logo} alt="movie disk logo" />
          <div className={s.separator}>|</div>
          <h1>PMDB</h1>
        </div>
      </div>
      <ScrollLink
        to="in-theaters"
        spy={false}
        smooth={true}
        offset={-50}
        duration={500}
        className={s.explore}
      >
        explore
        <p className={s.arrow} />
      </ScrollLink>
    </div>
  );
}
