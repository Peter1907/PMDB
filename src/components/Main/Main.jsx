/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import * as Scroll from 'react-scroll';
import { getPopMovies, getStoredPopMovies } from '../Redux/pop-movies/pop-movies';
import { getPopSeries, getStoredPopSeries } from '../Redux/pop-series/pop-series';
import { getStoredTopMovies, getTopMovies } from '../Redux/top-movies/top-movies';
import s from './Main.module.css';

const ScrollLink = Scroll.Link;

export default function Main() {
  const logo = ('./assets/logo02.png');

  const dispatch = useDispatch();
  const DATA = localStorage.getItem('TOP_MOVIES');
  const DATA2 = localStorage.getItem('POP_MOVIES');
  const DATA3 = localStorage.getItem('POP_SERIES');
  const Data = useSelector((state) => state.topMovies);
  const random = Math.floor(Math.random() * 247);
  const images = Data.slice(random, random + 2);

  const getData = () => {
    (DATA ? dispatch(getStoredTopMovies()) : dispatch(getTopMovies()));
    (DATA2 ? dispatch(getStoredPopMovies()) : dispatch(getPopMovies()));
    (DATA3 ? dispatch(getStoredPopSeries()) : dispatch(getPopSeries()));
  };

  useEffect(() => {
    getData();
  }, []);


  return (
    <div className={s.mainContainer}>
      <div className={s.imgContainer}>
        {images.map((item, id) => (
          <img key={id} id={`image-${id + 1}`} src={item.image} alt={images.title} />
        ))}
      </div>
      <div className={s.gradient} />
      <div className={s.textBox}>
        <div className={s.title}>
          <img src={logo} alt="movie disk logo" />
          <div className={s.separator}>|</div>
          <h1>PMDB</h1>
        </div>
        <h3 className={s.subTitle}>Media with Class</h3>
      </div>
      <ScrollLink
        to="in-theaters"
        spy={false}
        smooth={true}
        offset={-50}
        duration={500}
        className={s.explore}
      >
        explore
        <p className={s.arrow} />
      </ScrollLink>
    </div>
  );
}
