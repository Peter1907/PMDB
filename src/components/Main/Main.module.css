.mainContainer {
  position: relative;
  height: 100vh;
  overflow-y: hidden;
  background-color: #222831;
}

.background {
  width: 100vw;
  z-index: 1;
  position: fixed;
}

.gradient {
  background-image: linear-gradient(0.25turn, #222831, #222831, #222831e7, #2228315e, #22283100, #222831e8), linear-gradient(#22283100, #22283100, #22283100, #22283163, #222831);
  box-shadow: inset 0 0 100px #222831;
  width: 100vw;
  height: 100vh;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 5;
}

.textBox {
  position: absolute;
  top: 15%;
  left: 7%;
  z-index: 6;
}

.title {
  display: flex;
  align-items: center;
  gap: 1rem;
}

.title img {
  width: 120px;
}

.title h1 {
  font-size: 72px;
  color: #ffd369;
}

.separator {
  font-size: 100px;
  font-family: 'Cairo', sans-serif;
  font-weight: 200;
  margin: 0 0 1rem;
  color: #eee;
}

.subTitle {
  color: #eee;
  font-family: 'Cairo', sans-serif;
  font-size: 48px;
  font-weight: 300;
  margin: 0 0 0 8rem;
}

.imgContainer {
  display: grid;
  grid-template-columns: 1fr 1fr;
  max-width: 100%;
  position: absolute;
  right: 0;
  top: 0;
}

.imgContainer img {
  width: 100%;
}

.explore {
  z-index: 6;
  position: absolute;
  bottom: 2rem;
  left: 50%;
  transform: translate(-50%, 0);
  font-size: 46px;
  line-height: 1rem;
  color: #eee;
  font-family: 'Cairo', sans-serif;
  font-weight: 200;
  cursor: pointer;
}

.arrow {
  border-top: 2px solid #eee;
  border-right: 2px solid #eee;
  height: 2rem;
  width: 2rem;
  transform: rotate(135deg);
  align-self: flex-end;
  margin: 1rem auto 0;
}
