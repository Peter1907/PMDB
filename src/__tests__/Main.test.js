import renderer from 'react-test-renderer';
import Main from '../_mocks_/Main';

it('renders correctly', () => {
  const tree = renderer
    .create(<Main />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
