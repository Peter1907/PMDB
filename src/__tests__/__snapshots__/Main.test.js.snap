// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mainContainer"
>
  <div
    className="sliderContainer"
  >
    <div
      className="imgContainer"
    >
      <img
        id="image-1"
        src="https://m.media-amazon.com/images/M/MV5BMDFkYTc0MGEtZmNhMC00ZDIzLWFmNTEtODM1ZmRlYWMwMWFmXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_12,128,176_AL_.jpg"
      />
      <img
        id="image-2"
        src="https://m.media-amazon.com/images/M/MV5BM2MyNjYxNmUtYTAwNi00MTYxLWJmNWYtYzZlODY3ZTk3OTFlXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_12,128,176_AL_.jpg"
      />
    </div>
    <div
      className="dotsContainer"
    >
      <a
        href="image-0"
      >
         
      </a>
      <a
        href="image-1"
      >
         
      </a>
    </div>
  </div>
  <div
    className="gradient"
  />
  <div
    className="textBox"
  >
    <div
      className="title"
    >
      <img
        alt="movie disk logo"
        src="./assets/logo02.png"
      />
      <div
        className="separator"
      >
        |
      </div>
      <h1>
        PMDB
      </h1>
    </div>
  </div>
  <a
    className="explore"
    onClick={[Function]}
  >
    explore
    <p
      className="arrow"
    />
  </a>
</div>
`;
